RA Balance changes for March 2020

PunkPun opened the pull request:
### Changing all small-caliber weapons to instant hit
- Bullets take time to travel. The farther a target is, the less damage moving targets receive. As it stands most small caliber weapon ranges are small and unit movement speed is slow enough that the majority of units take full damage regardless of this complex system. This change makes all targets receive full damage regardless of movement and range. It mostly impacts Pillboxes and Mobile Flak anti-air capabilities. Bullets having speed is not mentioned anywhere in the game and very few people are even aware of it. My goal is to remove this unnecessary complexity
- Pillboxes are special because they shoot 6 bursts, but unlike all other assets in the game, they do not retarget in between them. It means that moving targets will still take less damage than stationary ones. The only difference is that damage will no longer depend on distance from the Pillbox to the target nor if the target is on roads. Rangers instead of taking no damage at maximum range, will take a little. Dogs instead of surviving the first pillbox volley, will die. That is about it
- Effect to Mobile Flak anti-air is by far the most significant. In summary Yaks and MiGs are fast, always moving and Mobile Flak has long anti-air range. All of this summed up means that MiGs do not even take damage at Flaks maximum range. Combined with SAM Site having very little effective range against MiGs makes it hard for Soviets to kill them. Whether they die mostly depends on the attentiveness of the person controlling them and not the efforts of the opponent to counter play. With instant-hit Flaks will now be able to properly counter. To substitute for this MiGs are getting a health buff (there is more info on it). This change is also more justifiable considering that AA Guns bullets had already been made faster a long time ago meaning they already have been dealing full damage

### Yak
- Loses opportunity fire
   - Currently Yaks are overperforming. Although the way Yak works is fun, it is really hard to balance them. They can not be at the same time a good sniper unit and an infantry mass murderer. As Yaks are planned to be reworked in the next +1 release, we decided not to try and mix those two roles together and just revert the change

### Longbow
- Loses opportunity fire
   - So it would be consistent between all aircraft. Opportunity fire does not really affect Longbow in meaningful ways
- Reload delay from 60 to 30
- Anti-air missiles
   - Speed from 384 to 492
   - Horizontal rate of turn from 10 to 25
   - Near enough from 0c298 to 0c596
   - Range from 5 to 4 cells
      - Longbows will no longer miss planes, once fired missiles instead of sometimes exploding due to bad tracking will always connect, they already did versus helicopters 
- Anti-ground missiles
   - Burst amount from 4 to 2
   - Burst delay from 14 to 7
      - Missiles will be burstier and will overkill less in big groups. Total ammo empties by about 1 second faster. This makes burstiness the same between air / ground weapons. Burst delay is done in a way that if Longbow fired at stationary target no ammo would be wasted
   - Damage from 6000 to 7000
   - Damage vs heavy from 100% to 125% 
      - Same damage as MiG, it will allow 1 Longbow to kill exactly 1 harvester or one Heavy Tank

Combined with the burst changes it will make Longbows much better at killing Mammoths from behind. Longbows currently are pretty mediocre for their cost. They can be very strong when you are ahead but are not able to carve an advantage in an equal game, even more so when you are losing

### MiG
- Health from 7500 to 10000
   - MiGs will take a lot more damage from Flaks because of the insta-hit changes, additionally Longbow gets an upgrade vs planes. It is a counter buff, additionally the damage from AA Guns make MiGs a very risky unit in SvA matchup. A health buff will make them more viable
   - It will be killed by 5 Rocket Soldier rockets instead of 4
- Loses opportunity fire
   - So it would be consistent between aircraft. Opportunity fire does not really affect MiG in meaningful ways

### Black Hawk 
- Loses opportunity fire
   - So it would be consistent between aircraft. Opportunity fire does not really affect BH in meaningful ways

### Chinook
- Becomes shared between both factions
   - It will allow for more varied map design and better naval gameplay. Plays we might see are Thief, Grenadier and Flamer drops. Thief looking like the most reliable out of the group, might become meta
Health from 14000 to 20000
   - They have lower vision than AA Guns have weapon range. This change increases the much needed time for reaction
   - As a consequence It can barely land 4 cells in front of a vision supported AA Gun for a one way trip. If you go deeper Chinook dies

### Turret 
- Damage from 6000 to 7500
   - This way in a Turret versus Tesla Coil battle both are equal, if Tesla is placed in front of a Turret, Tesla dies and vice versa. Turret will have slightly more dps than a Tesla

### Artillery 
- Damage vs heavy from 25% to 35%
   - It will take 5-6 shots to kill a defence instead of 7-8, It will have the same effective damage per second versus heavy armour as V2 Rocket Launcher. Ingame this number feels perfect, defences are not squishy but also are killed in a reasonable amount of time
Tesla Coil push is hard to fight back against, it requires quite a bit of pre-planning, knowledge and micro to counter. Allies struggle against it more than Soviets, this change along with the Turret one will slightly reduce that struggle

### Chrono Tank
- Rocket damage made the same as of Rocket Soldier
   - Damage from 2500 to 5000
   - Versus
      - none from 28% to 10%
      - wood from 88% to 74%
      - light from 88% to 34%
      - heavy from 120% to 100%
      - concrete from 60% to 50%
- Movement speed from 113 to 96
   - Medium Tank has Speed 82, Mobile Flak & Light Tank 118
      - Its counters will relatively become much faster than it. The portable chono ability will be the only way to outmaneuver its hunters. CTanks will have a harder time escaping after blinking aggressively 
- Health from 45000 to 40000
   - Instead of needing a second aircraft, you will need to use just 1 full ammo clip to kill it. This change was supposed to go with the Tesla Tank health nerf, but none of the Chrono Tank changes were implemented last release, they still lacked testing. This time with some minor adjustments we are more confident

Currently Chrono Tank has no role to fill, everything they do Allies can do better with other units, with these changes Chrono Tank becomes the best ground anti-armour unit. It deals incredible damage to heavy armour, can chase anything with the Portable Chrono ability, it receives reduced damage from most of the heavy armoured units as it has light armour itself, and a decent sized health pool.

Apart from aircraft Mobile Flaks are the best Soviet hard counter, although 1 Flak (700) barely loses to 1 Chrono Tank (1350) as the numbers of units rise the better Flaks become. They are much less bursty and can shoot while moving / turning giving them an edge in those engagements

Apart from aircraft Allies can counter with Light Tanks and Rangers, they do not trade as well as Flaks but still do it cost efficiently. It is important to focus fire CTanks with them. Additionally Allies have MRJs which can completely nullify CTank rockets

### Phase Transport
- Carry size from 4 to 5
   - To be more effective at dropping, it will not impact the game significantly
- Same damage buffs / nerfs as Chrono Tank as they both share the same weapon
- Splitting weapon from CTank and increasing reload delay from 60 to 100
   - In tests they were proven to be able to kill tanks cost efficiently in larger numbers. It is not the role this unit is supposed to have. 
   - Reload delay will be twice as much as Rocket Soldiers

### Minelayer 
- Will not receive damage from mines
   - When a Minelayer is caught laying mines it can be insta-killed as any shots to it will also damage mines beneath it. As of this change mines will explode but they will not kill the Minelayer. Current behaviour seems like a bug rather than a feature

Commit:
Add gdi08a

Commit:
Add gdi09ea

abcdefg30 committed:
Let VS2019 remove a duplicate line from the solution

abcdefg30 committed:
Add Ordos06a

pchote committed:
Remove IEquatable from ActorBoundsPair.

pchote committed:
Add support for polygon selection shapes.

pchote committed:
Move Interactable and Selectable to Mods.Common.

pchote committed:
Rework decoration renderable traits:

- Removed implicit pip definitions and IPips interface.
  New decoration traits have been added to render them.
  Pip types are no longer hardcoded in OpenRA.Game.

- Decoration rendering is now managed by SelectionDecorations(Base),
  which allows us to remove assumptions about the selection box
  geometry from the decoration traits.

- RenderNameTag has been replaced by WithNameTagDecoration, which is
  an otherwise normal decoration trait.

- Unify the configuration and reduce duplication between traits.

- Removed hardcoded references to specific selection box renderables.

- Remove legacy cruft.

pchote committed:
Update TD decorations.

pchote committed:
Update TS decorations.

dragunoff committed:
Align lobby bits in the player tab in TD

abcdefg30 committed:
Fix the settings tooltip container being overwritten ingame

PunkPun committed:
RA Balance patch

dragunoff committed:
Adjust the stroke of the muted indicator glyph

pchote committed:
Replace deprecated native OpenAL with OpenAL Soft on macOS.

pchote committed:
Update D2k decorations.

pchote committed:
Update RA decorations.

abcdefg30 committed:
Add support for an announcement function for carryall reinforcements

abcdefg30 committed:
Add support for destroying enemy carryalls

Mailaender committed:
Fix spy ignoring the target's faction.

pchote committed:
Remove invalid caching from GCOT.

pchote commented:
The soviet campaign missions will probably need to be updated to account for the chinook.

Mailaender committed:
Remove .lua scripts from the .NET solution file.

GraionDilach committed:
Add a shared parallel production queue.

pchote committed:
Implement isometric selection boxes for TS structures.

pchote committed:
Fix Waste Refinery bib.

pchote committed:
Remove special-case rollover rendering.

Mailaender committed:
Remove the hard-coded ban of placing buildings on resources.

pchote committed:
Remove text caching from CncLoadScreen.

We have repeatedly failed at invalidating these
cached values when things change, so the small perf
win is not worth the hassle.

pchote committed:
Move plugs and superweapons to Building queue.

pchote committed:
Disable plugs when there are no sockets to place them.

pchote committed:
Rename Defense button tooltip to Support.

Mailaender committed:
Remove unused terrain type.

pchote committed:
Fix detection circle line rendering.

pchote committed:
Restore ability to configure RMB orders + RMB panning.
